test/unit/uk/gov/hmrc/exports/services/reversemapping/declaration/items/SingleItemParserSpec.scala

```diff
@@ -33,7 +35,7 @@ class SingleItemParserSpec extends UnitSpec with EitherValues {
   override def beforeEach(): Unit = {
     super.beforeEach()
 
-    reset(procedureCodesParser)
+    when(procedureCodesParser.parse(any[NodeSeq])(any[MappingContext])).thenReturn(Right(None))
```

Comment:
It is still worth having `reset` method here, to reset mock's counters for e.g.

Reply:
But `beforeEach` is executed **before** each test, and `when` replaces any stub for that callsite. I cannot see how the counters could be different when a test starts. Still it's not a big deal to add the reset.

Reply:
This would work as you describe if you'd create a new stub before each test case. However, the stub instance is shared between test cases and only its behaviour is changed/overriden with `when` method. And this method does not reset the state of the mock, which means counters are not set to zero before each test.